public/src/components/Main.tsx

```diff
@@ -145,22 +154,59 @@ export class Main extends React.Component<MainProps, MainState> {
         }
     }
 
-    retryStream = (func: Function) => {
-        let newcounter = this.state.networkTimeOutCounter;
-        let newTimeout = this.state.networkTimeOut + Math.floor(newcounter/3) * this.state.networkTimeOut;
+    getStreamCounter = (flag: string) => {
+       if(flag === "notifications")
+          return this.state.networkTimeOutCounterNotifs;
+       else if(flag === "transactions")
+          return this.state.networkTimeOutCounterTrans;
+       else if(flag === "stockPrices")
+          return this.state.networkTimeOutCounterPrices;
+        else return -1;
+    }
 
-        if (newcounter == 2) {
-            const timeOut = this.state.networkTimeOut;
-            showErrorNotif("Unable to connect to server. Please check your internet connection. Retrying in " + (timeOut/1000) + "s", "Network error");
-            setTimeout(func, timeOut);
+    setStreamCounter = (flag: string, counter : number, func : Function) => {
+        if(flag === "notifications"){
+           this.setState({
+             networkTimeOutCounterNotifs:counter*2,
+           });
+           this.showRetryNotifs();
+           setTimeout(func,this.state.networkTimeOutCounterNotifs*this.state.networkTimeOut);
+         }
+        if(flag === "transactions"){
+          this.setState({
+            networkTimeOutCounterTrans:counter*2,
+          });
+          this.showRetryNotifs();
+          setTimeout(func,this.state.networkTimeOutCounterTrans*this.state.networkTimeOut);
         }
-
-        newcounter = (newcounter+1)%3;
+        if(flag === "stockPrices"){
         this.setState({
-            networkTimeOut: newTimeout,
-            networkTimeOutCounter: newcounter,
+          networkTimeOutCounterPrices:counter*2,
         });
-    };
+        this.showRetryNotifs();
+        setTimeout(func,this.state.networkTimeOutCounterPrices*this.state.networkTimeOut);
+      }
+    }
+
+    retryStream = (func: Function, flag: string) => {
+      this.setState({
+        connectionStatus: false,
+      });
+      const counter = this.getStreamCounter(flag);
+       if(counter != -1 && counter <= 1024)
+       {
+        this.setStreamCounter(flag,this.getStreamCounter(flag),func);
+      }
+    }
+
+    showRetryNotifs = () => {
+      if(this.state.networkTimeOutCounterNotifs === this.state.networkTimeOutCounterTrans && this.state.networkTimeOutCounterNotifs=== this.state.networkTimeOutCounterPrices)
```

Comment:
Explain this line? @ishanisri

Reply:
This is to show an error notification when all the three streams break , and when the 3 counters have the same value, otherwise it will lead to 3 notifications on screen for every incremented value of any counter